### PR TITLE
[settings] Hide episode thumb if unseen (core wise)

### PR DESCRIPTION
**Pull request by DaVukovic**
## Description

This is a similar change to #14914, but done in core. I also followed a suggestion to change the wordings for that setting to make it less confusing. To be pretty clear, and for the reason we have another option for the "Show plot/information"-setting, I also decided to change the option labels to show exactly what is meant. 

The only problem I can't solve myself (hence I added the "Help wanted" labe)l is, that the home screen TV show widgets don't change to the OverlaySpoiler image until the skin was reloaded (after playing a video, restarting Kodi, finish a database update, whatever). @Rechi helped me, but I don't get it ;). So I would love to see some help from someone who might be interested. 

## Motivation and Context

Same as for #14914

## How Has This Been Tested?

Tested on Ubuntu 16.04

## Screenshots (if appropriate):

See the PR mentioned above, The screenshots will be the same.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by DaVukovic** [comment]
@notspiff I mentioned you in case you are interested. If not, then sorry for the noise.

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -21588,3 +21588,22 @@ msgstr ""
...
+msgctxt "#39115"
+msgid "Movie plot"
+msgstr ""
+
+#. Label for "show information for unwatched items" setting option
+#: system/settings/settings.xml
+msgctxt "#39116"
+msgid "Episode plot"
+msgstr ""
+
```
Comment (Rechi) [inline addons/resource.language.en_gb/resources/strings.po]: translation script will remove this empty line

File: xbmc/video/VideoThumbLoader.cpp [hunk 2]
```diff
@@ -349,6 +351,21 @@ bool CVideoThumbLoader::LoadItemCached(CFileItem* pItem)
...
+  if (pItem->HasArt("thumb") && pItem->HasVideoInfoTag() &&
+      pItem->GetVideoInfoTag()->m_type == MediaTypeEpisode &&
+      pItem->GetVideoInfoTag()->GetPlayCount() == 0 &&
+      setting && 
+      !setting->FindIntInList(CSettings::VIDEOLIBRARY_THUMB_SHOW_UNWATCHED_EPISODE)
+     )
+  {
+    pItem->SetArt("thumb", "OverlaySpoiler.png");
+    m_videoDatabase->Close();
+    return true;
```
Comment (Rechi) [inline xbmc/video/VideoThumbLoader.cpp]: The exact same code is already after the if block.

File: addons/resource.language.en_gb/resources/strings.po (hunk 1 above)
Comment (DaVukovic) [reply addons/resource.language.en_gb/resources/strings.po]: Thought we need a newline at the end. Will remove

Comment (Rechi) [reply addons/resource.language.en_gb/resources/strings.po]: yes one, but not two

File: xbmc/video/VideoThumbLoader.cpp [hunk 3]
```diff
@@ -300,6 +301,7 @@ bool CVideoThumbLoader::LoadItem(CFileItem* pItem)
 
 bool CVideoThumbLoader::LoadItemCached(CFileItem* pItem)
 {
+
```
[the remark this answers is not in the document]
Comment (DaVukovic) [reply xbmc/video/VideoThumbLoader.cpp]: Ah, true. Added the include for another test and forgot to remove it again. Thanks

File: xbmc/video/VideoThumbLoader.cpp (hunk 2 above)
Comment (DaVukovic) [reply xbmc/video/VideoThumbLoader.cpp]: Wasn't sure, if I need to close the database. Will test removing that and update the PR if everything works fine (not arguing that it won't ;) ).

Comment (Rechi) [reply xbmc/video/VideoThumbLoader.cpp]: have look below, database is closed there too

File: addons/resource.language.en_gb/resources/strings.po (hunk 1 above)
Comment (DaVukovic) [reply addons/resource.language.en_gb/resources/strings.po]: Thought there was only one. Made sure, there is now

Comment (Rechi) [reply addons/resource.language.en_gb/resources/strings.po]: still two

**Comment by HitcherUK** [comment]
@DaVukovic RE: The home screen widgets not updating.

This is actually a problem with home screen widgets in general (see below), not just specific to your PR, so this is good go in my opinion.

To reproduce for other widgets simply find a movie in your library that's displayed on the home screen in Estuary, change the poster and go back home. Old poster is still shown until you either refresh the skin or play a video.

**Comment by DaVukovic** [comment]
@HitcherUK Yeah, I see what you mean. @Rechi PRd a hack to my branch, which I merged currently and I can confirm that announcing `OnScanFinished` at least refreshes the widgets. Also a skin reload would do that. I know that this is a hack and probably not the method we will do that in the future, but I can confirm it working. The widgets are refreshed fine now and the setting has an immediate effect.

**Comment by HitcherUK** [comment]
Thanks, this will make a lot of people happy.

**Comment by Rechi** [comment]
No still needs a change, but @DaVukovic knows what to change.

**Comment by DaVukovic** [comment]
squashed the commits and also renamed the topic for a better match ;)

File: xbmc/settings/MediaSettings.cpp [hunk 4]
```diff
@@ -343,6 +346,15 @@ void CMediaSettings::OnSettingAction(std::shared_ptr<const CSetting> setting)
   }
 }
 
+void CMediaSettings::OnSettingChanged(std::shared_ptr<const CSetting> setting)
+{
+  if (setting == nullptr)
+    return;
+
+  if (setting->GetId() == CSettings::SETTING_VIDEOLIBRARY_SHOWUNWATCHEDPLOTS)
+    CServiceBroker::GetAnnouncementManager()->Announce(ANNOUNCEMENT::VideoLibrary, "xbmc", "OnScanFinished");
```
Comment (Rechi) [inline xbmc/settings/MediaSettings.cpp]: Either change this to `OnUpdate` or add a new type at the following location.
https://github.com/xbmc/xbmc/blob/05611ab739c8f0672ffd0983cdb01d7ed8a5caa8/xbmc/listproviders/DirectoryProvider.cpp#L255-L258

Comment (DaVukovic) [reply xbmc/settings/MediaSettings.cpp]: Done that, build it and it also works. Decided to use `OnUpdate`

thx

**Review by Rechi — APPROVED** [review]

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by HitcherUK** [comment]
@MartijnKaijser I believe this is good to go now.

**Pull request merged**